lib/tether/ops_linux.go

```diff
@@ -425,23 +425,34 @@ func (t *BaseOperations) Apply(endpoint *NetworkEndpoint) error {
 }
 
 func apply(nl Netlink, t *BaseOperations, endpoint *NetworkEndpoint) error {
+	if endpoint.applied {
+		log.Infof("skipping applying config for network %s as it has been applied already", endpoint.Network.Name)
+		return nil // already applied
+	}
+
 	// Locate interface
 	slot, err := strconv.Atoi(endpoint.ID)
 	if err != nil {
-		detail := fmt.Sprintf("endpoint ID must be a base10 numeric pci slot identifier: %s", err)
-		return errors.New(detail)
+		return fmt.Errorf("endpoint ID must be a base10 numeric pci slot identifier: %s", err)
 	}
-	link, err := nl.LinkBySlot(int32(slot))
+
+	defer func() {
```

Comment:
I really dislike this pattern when deferred function checks for errors and set some flags, it should be done by a higher level caller. I see benefits of this approach, I understand the reason, it just makes this code so hard to follow. I needed to unroll so much code to ensure I am not missing anything. This function has to be already split into multiple smaller chunks to make it simpler.